src/wasi/Wasi.cpp

```diff
@@ -64,6 +64,12 @@ void WASI::proc_exit(ExecutionState& state, Value* argv, Value* result, Instance
     ASSERT_NOT_REACHED();
 }
 
+void WASI::proc_raise(ExecutionState& state, Value* argv, Value* result, Instance* instance)
+{
+    ASSERT(argv[0].type() == Value::I32);
```

Comment:
You do not need to check for type, as `asI32()` will check it and it will crash anyway if the type is incorrect.

Reply:
Thanks! :)